java/core/src/test/java/co/worklytics/psoxy/RulesTest.java

```diff
@@ -0,0 +1,75 @@
+package co.worklytics.psoxy;
+
+import co.worklytics.test.TestUtils;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.dataformat.yaml.YAMLFactory;
+import lombok.SneakyThrows;
+import org.junit.jupiter.api.Test;
+
+import java.util.regex.Pattern;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RulesTest {
+
+
+    final String EXAMPLE_YAML =
+        "pseudonymizations:\n" +
+        "- relativeUrlRegex: \"/calendar/v3/calendars/.*/events.*\"\n" +
+        "  jsonPaths:\n" +
+        "  - \"$..email\"\n" +
+        "emailHeaderPseudonymizations: []\n" +
+        "redactions:\n" +
+        "- relativeUrlRegex: \"/calendar/v3/calendars/.*/events.*\"\n" +
+        "  jsonPaths:\n" +
+        "  - \"$..displayName\"\n" +
+        "  - \"$.items[*].extendedProperties.private\"\n";
+
+    Rules rules = Rules.builder().pseudonymization(Rules.Rule.builder()
+            .relativeUrlRegex("/calendar/v3/calendars/.*/events.*")
+            .jsonPath("$..email")
+            .build())
+        .redaction(Rules.Rule.builder()
+            .relativeUrlRegex("/calendar/v3/calendars/.*/events.*")
+            .jsonPath("$..displayName")
+            .jsonPath("$.items[*].extendedProperties.private")
+            .build())
+        .build();
+
+
+    @SneakyThrows
+    @Test
+    public void yaml() {
+        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+
+        assertEquals("---\n" + EXAMPLE_YAML, //q: why does it add this prefix??
+            objectMapper.writeValueAsString(rules));
+
+        Rules fromYaml = objectMapper.readerFor(Rules.class)
+            .readValue(EXAMPLE_YAML);
+
+        Pattern pattern = Pattern.compile(fromYaml.getPseudonymizations().get(0).getRelativeUrlRegex());
+
+        assertFalse(pattern.matcher("/obviously-not-the-url").matches());
+        assertTrue(pattern.matcher("/calendar/v3/calendars/primary/events/1248asdfas4532").matches());
+    }
+
+    @SneakyThrows
+    @Test
+    public void yaml_file() {
+        byte[] data = TestUtils.getData("rules/example.yaml");
+
+        ObjectMapper objectMapper = new ObjectMapper(new YAMLFactory());
+
+        Rules fromYaml = objectMapper.readerFor(Rules.class)
```

Comment:
this test is basically the same: is not testing any rule but just that the object mapper can read from bytes as opposed to String like in the first test.

Reply:
yeah, and just that no encoding issues cause weirdness with the regexes.